plugins/package-managers/pub/src/test/kotlin/PubspecTest.kt

```diff
@@ -0,0 +1,135 @@
+                  pkg-1: ^1.0.0
+                  pkg-2:
+                    hosted: https://some-package-server.com
+                    version: ^1.4.0
+                  pkg-3:
+                    hosted:
+                      name: pkg-3
+                      url: https://some-other-package-server.com
+                    version: ^1.4.0  
+            """.toPubspecYaml()
+
+            parsePubspec(yaml).dependencies shouldBe mapOf(
+                "pkg-1" to HostedDependency(
+                    version = "^1.0.0"
+                ),
+                "pkg-2" to HostedDependency(
+                    url = "https://some-package-server.com",
+                    version = "^1.4.0"
+                ),
+                "pkg-3" to HostedDependency(
+                    url = "https://some-other-package-server.com",
+                    version = "^1.4.0"
+                )
+            )
+        }
+
+        "parse SDK dependencies" {
+            val yaml = """
+                dependencies:
+                  flutter_driver:
+                    sdk: flutter
+            """.toPubspecYaml()
+
+            parsePubspec(yaml).dependencies shouldBe mapOf(
+                "flutter_driver" to SdkDependency(
+                    sdk = "flutter"
+                )
+            )
+        }
+
+        "parse path dependencies" {
+            val yaml = """
+                dependencies:
+                  transmogrify:
+                    path: /Users/me/transmogrify
+            """.toPubspecYaml()
+
+            parsePubspec(yaml).dependencies shouldBe mapOf(
+                "transmogrify" to PathDependency(
+                    path = "/Users/me/transmogrify"
+                )
+            )
+        }
+
+        "parse git dependencies" {
+            val yaml = """
+                dependencies:
+                  pkg-1:
+                    git: git@github.com:some-org/some-project.git
+                  pkg-2:
+                    git:
+                      url: git@github.com:some-org/some-project.git
+                      path: path/to/pkg-2
+                  pkg-3:
+                    git:
+                        url: git@github.com:some-org/some-project.git
+                        ref: some-branch
+            """.toPubspecYaml()
+
+            parsePubspec(yaml).dependencies shouldBe mapOf(
+                "pkg-1" to GitDependency(
+                    url = "git@github.com:some-org/some-project.git"
+                ),
+                "pkg-2" to GitDependency(
+                    url = "git@github.com:some-org/some-project.git",
+                    path = "path/to/pkg-2"
+                ),
+                "pkg-3" to GitDependency(
+                    url = "git@github.com:some-org/some-project.git",
+                    ref = "some-branch"
+                )
+            )
+        }
+    }
+})
+
+private fun String.toPubspecYaml() =
+    buildString {
+        appendLine("name: some-package")
```

Comment:
If I'm not mistaken this function only saves you to add line single line to all call sites. Does that really pay off, or should we inline this to avoid the need for readers of the code to lookup what the codes actually does?

Reply:
I believe it makes sense as-is, to make the test focus only on the relevant bits.